common/service/dynamicconfig/config.go

```diff
@@ -21,111 +21,88 @@
-	return func(filterVal string) interface{} {
-		filters := make(map[Filter]interface{})
-		filters[filter] = filterVal
-		val, err := c.client.GetValueWithFilters(key, filters)
+// GetProperty gets a eface property and returns defaultValue if property is not found
+func (c *Collection) GetProperty(key Key, defaultValue interface{}) func() interface{} {
+	return func() interface{} {
+		val, err := c.client.GetValue(key, defaultValue)
 		if err != nil {
-			return defaultVal
+			c.logNoValue(key, err)
 		}
 		return val
 	}
 }
 
-// GetProperty gets a eface property and returns defaultVal if property is not found
-func (c *Collection) GetProperty(key Key, defaultVal interface{}) func() interface{} {
-	return func() interface{} {
-		val, err := c.client.GetValue(key)
-		if err != nil {
-			return defaultVal
-		}
-		return val
+func getFilterMap(opts ...FilterOption) map[Filter]interface{} {
+	l := len(opts)
+	m := make(map[Filter]interface{}, l)
+	for _, opt := range opts {
+		opt(m)
 	}
+	return m
 }
 
 // GetIntProperty gets property and asserts that it's an integer
-func (c *Collection) GetIntProperty(key Key, defaultVal int) func() int {
-	return func() int {
-		return c.GetProperty(key, defaultVal)().(int)
+func (c *Collection) GetIntProperty(key Key, defaultValue int) func(opts ...FilterOption) int {
+	return func(opts ...FilterOption) int {
+		val, err := c.client.GetIntValue(key, getFilterMap(opts...), defaultValue)
+		if err != nil {
+			c.logNoValue(key, err)
+		}
+		return val
 	}
 }
 
 // GetFloat64Property gets property and asserts that it's a float64
-func (c *Collection) GetFloat64Property(key Key, defaultVal float64) func() float64 {
-	return func() float64 {
-		return c.GetProperty(key, defaultVal)().(float64)
+func (c *Collection) GetFloat64Property(key Key, defaultValue float64) func(opts ...FilterOption) float64 {
+	return func(opts ...FilterOption) float64 {
+		val, err := c.client.GetFloatValue(key, getFilterMap(opts...), defaultValue)
+		if err != nil {
+			c.logNoValue(key, err)
+		}
+		return val
 	}
 }
 
 // GetDurationProperty gets property and asserts that it's a duration
-func (c *Collection) GetDurationProperty(key Key, defaultVal time.Duration) func() time.Duration {
-	return func() time.Duration {
-		return c.GetProperty(key, defaultVal)().(time.Duration)
+func (c *Collection) GetDurationProperty(key Key, defaultValue time.Duration) func(opts ...FilterOption) time.Duration {
+	return func(opts ...FilterOption) time.Duration {
+		val, err := c.client.GetDurationValue(key, getFilterMap(opts...), defaultValue)
+		if err != nil {
+			c.logNoValue(key, err)
+		}
+		return val
 	}
 }
 
 // GetBoolProperty gets property and asserts that it's an bool
-func (c *Collection) GetBoolProperty(key Key, defaultVal bool) func() bool {
-	return func() bool {
-		return c.GetProperty(key, defaultVal)().(bool)
+func (c *Collection) GetBoolProperty(key Key, defaultValue bool) func(opts ...FilterOption) bool {
```

Comment:
Can we define types for return type of each kind of PropertyValue?  Like BoolPropertyValue in this case?

Reply:
Done.